tests/cli/main.rs

```diff
@@ -274,6 +274,119 @@ fn basic_backup() {
     // TODO: Compare vs source tree.
 }
 
+#[test]
+#[cfg(unix)]
+fn backup_user_and_permissions() {
```

Comment:
It would be nice to make this test stronger. I suspect it might almost pass even if the owners and perms are not stored...

Testing changes to the owner and group may be difficult in a hermetic way. Especially if the tests aren't running as root we probably cannot set the user.

We could look at what groups the test runner belongs to, and change some files to be owned by different groups? There is very likely more than one? (Maybe not in CI?)

For permissions, it would be nice to make some files executable, readonly, and group-writeable, and check that the permissions are restored.

In particular there could conceivably be a bug where a restored file is made readonly too soon and so can't be written. It'd be nice to have a test showing that readonly files are properly restored.

Reply:
Agreed, this test should be more robust.

Reply:
Added the test `backup_unix_permissions` which should properly test the unix permissions. Unsure how to test the user/group so I'm just gonna hold off on that for now.

Reply:
I had a look at this and extended it a bit, and I think it's OK for now. More thoroughly testing this probably requires some scaffolding to save indexes that don't exist on disk. That'd be good to have but it's well beyond the scope of this PR.